sstables/sstable_directory.cc

```diff
@@ -441,9 +441,9 @@ sstable_directory::filter_sstables(std::function<future<bool>(sstables::shared_s
 template <std::ranges::range Container, typename Func>
 requires std::is_invocable_r_v<future<>, Func, typename std::ranges::range_value_t<Container>&>
 future<>
-sstable_directory::parallel_for_each_restricted(Container& c, Func func) {
-    co_await max_concurrent_for_each(c, _manager.dir_semaphore()._concurrency, [&] (auto& el) -> future<>{
-        auto units = co_await get_units(_manager.dir_semaphore()._sem, 1);
+sstable_directory::parallel_for_each_restricted(sstables_manager& manager, Container& c, Func func) {
+    co_await max_concurrent_for_each(c, manager.dir_semaphore()._concurrency, [&] (auto& el) -> future<>{
+        auto units = co_await get_units(manager.dir_semaphore()._sem, 1);
```

Comment:
It looks like manager only exports its dir_semaphore for this particular method and that's it. Maybe it means that the `parallel_for_each_restricted(sstables_manager&, ...)` should be manager's method, not sstable_directory's one?

Reply:
Makes sense.  It's in sstable_directory for historical reasons.
When it was introduced, IIRC, the dir semaphore was allocated in main and the sstable_directory was handed a reference to it. (See https://github.com/scylladb/scylladb/blob/4d6739c2e6b395d171acf4d17734ef3b43ff9e85/main.cc#L503)